sklearn/preprocessing/target.py

```diff
@@ -0,0 +1,242 @@
+
+    inverse_func : function, optional
+        Function to apply to the prediction of the regressor. Cannot be set at
+        the same time than ``transformer``. If ``None`` and ``transformer`` as
+        well, the function used will be the identity function. The inverse
+        function is used to return to the same space of the original training
+        labels during prediction.
+
+    check_inverse : bool, (default=True)
+        Whether to check that ``transform`` followed by ``inverse_transform``
+        or ``func`` followed by ``inverse_func`` leads to the original data.
+
+    Attributes
+    ----------
+    regressor_ : object
+        Fitted regressor.
+
+    transformer_ : object
+        Used transformer in ``fit`` and ``predict``.
+
+    y_ndim_ : int
+        Number of targets.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.linear_model import LinearRegression
+    >>> from sklearn.preprocessing import TransformedTargetRegressor
+    >>> tt = TransformedTargetRegressor(regressor=LinearRegression(),
+    ...                                 func=np.log, inverse_func=np.exp)
+    >>> X = np.arange(4).reshape(-1, 1)
+    >>> y = np.exp(2 * X).ravel()
+    >>> tt.fit(X, y)
+    ... #doctest: +NORMALIZE_WHITESPACE
+    TransformedTargetRegressor(check_inverse=True,
+                               func=<ufunc 'log'>,
+                               inverse_func=<ufunc 'exp'>,
+                               regressor=LinearRegression(copy_X=True,
+                                                          fit_intercept=True,
+                                                          n_jobs=1,
+                                                          normalize=False),
+                               transformer=None)
+    >>> tt.score(X, y)
+    1.0
+    >>> tt.regressor_.coef_
+    array([ 2.])
+
+    """
+    def __init__(self, regressor=LinearRegression(), transformer=None,
+                 func=None, inverse_func=None, check_inverse=True):
+        self.regressor = regressor
+        self.transformer = transformer
+        self.func = func
+        self.inverse_func = inverse_func
+        self.check_inverse = check_inverse
+        # we probably need to change this ones we have tags
+        self._estimator_type = regressor._estimator_type
+
+    def _fit_transformer(self, y, sample_weight):
+        if (self.transformer is not None and
+                (self.func is not None or self.inverse_func is not None)):
+            raise ValueError("Both 'transformer' and functions 'func'/"
+                             "'inverse_func' cannot be set at the same time.")
+        elif self.transformer is not None:
+            self.transformer_ = clone(self.transformer)
+        else:
+            self.transformer_ = FunctionTransformer(
+                func=self.func, inverse_func=self.inverse_func, validate=False)
+        fit_parameters = signature(self.transformer_.fit).parameters
```

Comment:
In fact, I could not find a transformer with `sample_weight`. I replicated the the code that we have in the bagging estimator:

```python
support_sample_weight = has_fit_parameter(self.regressor_, 'sample_weight')
if support_sample_weight:
    if sample_weight is None:
        current_sample_weight = np.ones((y.shape[0],))
    else: 
        current_sample_weight = sample_weight
```
In case that transformers should handle `sample_weight`, it will be a use case for the `sample_prop` since several `sample_weight` could be passed to the regressor and to the transformer.